src/cryptography/x509/extensions.py

```diff
@@ -733,6 +733,46 @@ class OCSPNoCheck(object):
 
 
 @utils.register_interface(ExtensionType)
+class TLSFeature(object):
+    oid = ExtensionOID.TLS_FEATURE
+
+    def __init__(self, features):
+        features = list(features)
+        if not all(isinstance(x, TLSFeatureType) for x in features):
+            raise TypeError(
+                "features must be a list of elements from the TLSFeatureType "
+                "enum"
+            )
+
+        self._features = features
+
+    def __repr__(self):
+        return "<TLSFeature(features={0.features})>".format(self)
+
+    def __eq__(self, other):
+        if not isinstance(other, TLSFeature):
+            return NotImplemented
+
+        return self.features == other.features
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __hash__(self):
+        return hash(tuple(self.features))
+
+    features = utils.read_only_property("_features")
```

Comment:
hmm, so conventionally we make the class iterable/subscriptable, instead of exposing the list as an attr